PIPELINE.md

```diff
@@ -0,0 +1,109 @@
+submitted to the source repository, a new OpenShift project is created to build and test the
+merged PR and allow for manual testing. Once the manual testing is approved, the build succeeds.
+
+### GitHub Setup
+
+1. Create your own forks of [nationalparks](https://github.com/openshift-roadshow/nationalparks.git), 
+   [mlbparks](https://github.com/openshift-roadshow/mlbparks.git), and 
+   [parksmap-web](https://github.com/openshift-roadshow/parksmap-web.git).
+2. Rename each of those repositories in GitHub with a common prefix, for example:
+   `myapp-nationalparks`, `myapp-mlbparks`, `myapp-parksmap`.
+3. Create a GitHub personal access token with access to `repo`, `admin:repo_hook`, `admin:org_hook`.
+   (Keep your token to use below as GITHUB-TOKEN).
+
+### OpenShift/Jenkins Setup
+
+1. Create a new project oc new-project PROJECT-NAME
+2. Assign the right permission to your jenkins service account:
+
+   ```
+   oc login -u system:admin
+   oc adm policy add-cluster-role-to-user self-provisioner system:service-account:PROJECT-NAME:jenkins
+   oc login -u developer
+   ```
+
+3. Add Jenkins to your project: 
+
+   ```
+   oc new-app openshift/jenkins-ephemeral -p MEMORY_LIMIT=1Gi
+   ```
+
+4. Expose the jenkins service externally (for GitHub webhooks) with ngrok: 
+
+   ```
+   oc new-app -f https://raw.githubusercontent.com/csrwng/ngrok/master/openshift/ngrok-template.yaml -p HOST=jenkins -p PORT=80
+   ```
+
+5. Get public webhook URL from ngrok: Navigate to the ngrok application. The main page will display 2 URLs. Copy either URL to use for your Jenkins webhook.
+6. Configure Github Server on Jenkins
+    1. Login to Jenkins
+    2. Go to Manage Jenkins -> Configure System
+    3. Under GitHub section:
+        1. Click on the `Advanced...` button
+        2. Check `Specify another hook url for GitHub configuration`
+        3. Replace the default URL host with the one from ngrok (url should look something like: http://47e8e09b.ngrok.io/github-webhook/)
+        4. Click on `Add Github Server` and select `GitHub Server`
+        5. Click on `Add` next to `Credentials` and select `Jenkins`
+        6. Leave `Kind` as `Username with password`. For `Username` enter your GitHub username, for `Password` enter your GITHUB-TOKEN, click on `Add`
+        7. Click on `Add` again (as in step 5)
+        8. Select a `Kind` of `Secret text`. For `Secret` paste your GITHUB-TOKEN, and for `Description` enter descriptive text like `Github Token`, click on `Add`
+        9. Under the `Credentials` dropdown, select the one you just created.
+        10. Click on `Test connection`
+    4. Click on `Save` at the bottom of the screen.
+7. Add Github Organization Plugin
+    1. Click on Manage Jenkins -> Manage Plugins -> Available
+    2. In the `Filter` search box, enter `github organization`
+    3. Check the box next to `GitHub Organization Folder Plugin`
+    4. Click on `Install without restart`
+    5. Check the checkbox next to `Restart Jenkins...` 
```

Comment:
check or uncheck?  it's weird that step 4 is "install w/o restart" and step 5 is "check the restart jenkins checkbox".

guessing the latter is some sort of soft re-init but it's still confusing.